ansible/roles/passwords/tasks/main.yml

```diff
@@ -7,6 +7,19 @@
   delegate_to: localhost
   run_once: true
 
+- name: Generate k3s token
+  ansible.builtin.set_fact:
+    k3s_token_secret: ""
+
+- name:  Generate k3s token and add to terraform
```

Comment:
If you see https://developer.hashicorp.com/terraform/language/values/variables#assigning-values-to-root-module-variables, you will see that any vars file ending `.auto.tfvars.json` or `.auto.tfvars` gets automatically loaded.

So rather than regex replacing something in the git-controlled .tfvars file, lets instead *template* a file here like e.g. `ansible-provided.auto.tfvars.json` and template something out using ` ... | to_nice_json` ?

That provides a more general mechanism to set terraform vars from ansible.

Use json so you don't have to template HCL.